### PR TITLE
Fixes to the compilation tests after (u)int - (u)int256 separation

### PR DESCRIPTION
**Pull request by theo25**
This PR fixes the compilation tests that were failing to compile due to type error related with the separation of the `(u)int` and `(u)int256` types. Generally speaking the fix constitutes of replaces instances of `(u)int` that used to be syntactic sugar with `(u)int256`. Occasionally there was also a need to cast the result of evaluating the `length` member of arrays to `uint256`, since its type has changed to `uint`.

All compilation tests can now pass the front end and fail in the same way they did before the separation (commit 3499b3f).

### CODE REVIEW

File: test/compilationTests/gnosis/MarketMakers/LMSRMarketMaker.sol [hunk 1]
```diff
@@ -86,12 +86,12 @@ contract LMSRMarketMaker is MarketMaker {
...
         constant
-        returns (uint price)
+        returns (uint256 price)
     {
         require(market.eventContract().getOutcomeCount() > 1);
-        int[] memory netOutcomeTokensSold = getNetOutcomeTokensSold(market);
-        int logN = Math.ln(netOutcomeTokensSold.length * ONE);
-        uint funding = market.funding();
+        int256[] memory netOutcomeTokensSold = getNetOutcomeTokensSold(market);
+        int256 logN = Math.ln(uint8(netOutcomeTokensSold.length) * ONE);
```
Comment (dwightguth) [inline test/compilationTests/gnosis/MarketMakers/LMSRMarketMaker.sol]: should be uint256

File: test/compilationTests/gnosis/MarketMakers/LMSRMarketMaker.sol [hunk 2]
```diff
@@ -22,25 +22,25 @@ contract LMSRMarketMaker is MarketMaker {
...
+        returns (uint256 cost)
     {
         require(market.eventContract().getOutcomeCount() > 1);
-        int[] memory netOutcomeTokensSold = getNetOutcomeTokensSold(market);
+        int256[] memory netOutcomeTokensSold = getNetOutcomeTokensSold(market);
         // Calculate cost level based on net outcome token balances
-        int logN = Math.ln(netOutcomeTokensSold.length * ONE);
-        uint funding = market.funding();
-        int costLevelBefore = calcCostLevel(logN, netOutcomeTokensSold, funding);
+        int256 logN = Math.ln(uint8(netOutcomeTokensSold.length) * ONE);
```
Comment (dwightguth) [inline test/compilationTests/gnosis/MarketMakers/LMSRMarketMaker.sol]: should be uint256

File: test/compilationTests/gnosis/MarketMakers/LMSRMarketMaker.sol [hunk 3]
```diff
@@ -57,26 +57,26 @@ contract LMSRMarketMaker is MarketMaker {
...
+        returns (uint256 profit)
     {
         require(market.eventContract().getOutcomeCount() > 1);
-        int[] memory netOutcomeTokensSold = getNetOutcomeTokensSold(market);
+        int256[] memory netOutcomeTokensSold = getNetOutcomeTokensSold(market);
         // Calculate cost level based on net outcome token balances
-        int logN = Math.ln(netOutcomeTokensSold.length * ONE);
-        uint funding = market.funding();
-        int costLevelBefore = calcCostLevel(logN, netOutcomeTokensSold, funding);
+        int256 logN = Math.ln(uint8(netOutcomeTokensSold.length) * ONE);
```
Comment (dwightguth) [inline test/compilationTests/gnosis/MarketMakers/LMSRMarketMaker.sol]: should be uint256

**Comment by theo25** [comment]
All of them are fixed now.

**Pull request merged**